ecs-agent/api/errors/errors.go

```diff
@@ -43,14 +46,22 @@ func IsInstanceTypeChangedError(err error) bool {
 	if awserr, ok := err.(awserr.Error); ok {
 		return strings.Contains(awserr.Message(), InstanceTypeChangedErrorMessage)
 	}
+
+	var apiErr smithy.APIError
+	if errors.As(err, &apiErr) {
+		return strings.Contains(apiErr.ErrorMessage(), InstanceTypeChangedErrorMessage)
+	}
+
```

Comment:
this should go inside the ok {} block above right? why is it a separate check ? 

same applies to IsClusterNotFoundError below.